enterprise/server/backends/distributed/distributed_test.go

```diff
@@ -263,20 +263,20 @@ func TestEventualConsistency(t *testing.T) {
 			i := rand.Intn(len(peers))
 			return i, peers[i]
 		}
-		useRandomCache := func(fn func(c *Cache)) {
+		useRandomCache := func(fn func(c *Cache, peer string)) {
 			mu.Lock()
 			defer mu.Unlock()
 			_, p := randomPeer()
-			fn(caches[p])
+			fn(caches[p], p)
 		}
 		shutdownRandomCache := func() string {
 			i, p := randomPeer()
 			log.Printf("Shutting down %q...", p)
 			mu.Lock()
-			defer mu.Unlock()
```

Comment:
Moving this lock means that the getting/setting test now resumes as soon as we've mutated the peers list / cache set, but *before* the cache has fully shut down. This makes the test less lenient, because now we are actively sending remote-fetch requests to the node which is shutting down, relying on the coordinator to correctly handle this.